src/oc/web/actions/activity.cljs

```diff
@@ -321,47 +350,59 @@
-          (api/create-board final-board-data (:invite-note fixed-edited-data)
-            (fn [{:keys [success status body] :as response}]
-              (if (= status 409)
-                ;; Board name exists
-                (board-name-exists-error fixed-edit-key)
-                (create-update-entry-cb fixed-edited-data fixed-edit-key response)))))
-        (let [org-slug (router/current-org-slug)
-              entry-board-data (dis/board-data @dis/app-state org-slug (:board-slug fixed-edited-data))
-              entry-create-link (utils/link-for (:links entry-board-data) "create")]
-          (api/create-entry fixed-edited-data fixed-edit-key entry-create-link create-update-entry-cb))))
-    (dis/dispatch! [:entry-save fixed-edit-key])))
+(defn entry-save
+  ([edit-key edited-data section-editing]
+     (entry-save edit-key edited-data section-editing create-update-entry-cb))
+
+  ([edit-key edited-data section-editing entry-save-cb]
+     (let [fixed-edited-data (assoc edited-data :status (or (:status edited-data) "draft"))
+           fixed-edit-key (or edit-key :entry-editing)]
+       (if (:links fixed-edited-data)
+         (if (and (= (:board-slug fixed-edited-data) utils/default-section-slug)
+                  section-editing)
+           (let [fixed-entry-data (dissoc fixed-edited-data :board-slug :board-name :invite-note)
+                 final-board-data (assoc section-editing :entries [fixed-entry-data])]
+             (api/create-board final-board-data (:invite-note fixed-edited-data)
+               (fn [{:keys [success status body] :as response}]
+                 (if (= status 409)
+                   ;; Board name exists
+                   (board-name-exists-error fixed-edit-key)
+                   (entry-save-cb fixed-edited-data fixed-edit-key response)))))
+           (api/update-entry fixed-edited-data fixed-edit-key entry-save-cb))
+         (if (and (= (:board-slug fixed-edited-data) utils/default-section-slug)
+                  section-editing)
+           (let [fixed-entry-data (dissoc fixed-edited-data :board-slug :board-name :invite-note)
+                 final-board-data (assoc section-editing :entries [fixed-entry-data])]
+             (api/create-board final-board-data (:invite-note fixed-edited-data)
+               (fn [{:keys [success status body] :as response}]
+                 (if (= status 409)
+                   ;; Board name exists
+                   (board-name-exists-error fixed-edit-key)
+                   (entry-save-cb fixed-edited-data fixed-edit-key response)))))
+           (let [org-slug (router/current-org-slug)
+                 entry-board-data (dis/board-data @dis/app-state org-slug (:board-slug fixed-edited-data))
+                 entry-create-link (utils/link-for (:links entry-board-data) "create")]
+             (api/create-entry fixed-edited-data fixed-edit-key entry-create-link entry-save-cb))))
+       (dis/dispatch! [:entry-save fixed-edit-key]))))
```

Comment:
This all would be good to have some comments at least to understand what's doing in the various cases (create new entry, create new entry with new board, update existing entry etc..).
I know it's not your code, you can avoid it if you don't have enough time..